sweepbatcher/presigned.go

```diff
@@ -0,0 +1,602 @@
+package sweepbatcher
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/btcsuite/btcd/blockchain"
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightningnetwork/lnd/lntypes"
+	"github.com/lightningnetwork/lnd/lnwallet/chainfee"
+)
+
+// ensurePresigned checks that we can sign a transaction sweeping the inputs of
+// this group only. This method is called when a group of sweeps of new swap is
+// being added to the batch. If the batch can be either empty or existing. This
```

Comment:
nit: comment seems off

Reply:
Fixed.

```go
// ensurePresigned checks there there is a presigned transaction spending the
// inputs of this group only. If allowNonEmptyBatch is false, the batch must be
// empty.
func (b *batch) ensurePresigned(ctx context.Context, newSweeps []*sweep,
        allowNonEmptyBatch bool) error {
```

Also I factored out pure function `ensurePresigned` from the method and added a unit test for it.